src/signing.rs

```diff
@@ -671,24 +671,13 @@ impl Drop for ExpandedSecretKey {
 
 impl From<&SecretKey> for ExpandedSecretKey {
     fn from(secret_key: &SecretKey) -> ExpandedSecretKey {
-        let mut h: Sha512 = Sha512::default();
-        let mut hash: [u8; 64] = [0u8; 64];
-        let mut lower: [u8; 32] = [0u8; 32];
-        let mut upper: [u8; 32] = [0u8; 32];
-
-        h.update(secret_key);
-        hash.copy_from_slice(h.finalize().as_slice());
-
-        lower.copy_from_slice(&hash[00..32]);
-        upper.copy_from_slice(&hash[32..64]);
-
-        lower[0] &= 248;
-        lower[31] &= 63;
-        lower[31] |= 64;
+        let hash = Sha512::default().chain_update(secret_key).finalize();
+        let (lower, upper) = hash.split_at(32);
```

Comment:
Maybe leave a comment to use `array_chunks` when stable to get rid of the `unwrap`s below?

(might also be good to add a `clippy::unwrap_used` lint too)

Reply:
What do you mean add the lint? As in make it a global Deny, and then do an explicit Allow in this function?

Reply:
Yeah